share/getters/cascade.go

```diff
@@ -122,16 +121,18 @@ func cascadeGetters[V any](
 		getCtx, cancel := ctxWithSplitTimeout(ctx, len(getters)-i, 0)
 		val, getErr := get(getCtx, getter)
 		cancel()
-		if getErr == nil {
-			return val, nil
+		if getErr == nil || errors.Is(getErr, share.ErrNamespaceNotFound) {
+			return val, getErr
 		}
-		if errors.Is(share.ErrNamespaceNotFound, getErr) {
-			return zero, getErr
+
+		if errors.Is(getErr, errOperationNotSupported) {
+			continue
 		}
 
-		if !errors.Is(getErr, errOperationNotSupported) {
-			err = errors.Join(err, getErr)
-			span.RecordError(getErr, trace.WithAttributes(attribute.Int("getter_idx", i)))
+		err = errors.Join(err, getErr)
+		span.RecordError(getErr, trace.WithAttributes(attribute.Int("getter_idx", i)))
+		if ctx.Err() != nil {
+			return zero, err
```

Comment:
```suggestion
			return zero, ctx.Err()
```

Reply:
We return aggregated error, because there could be legit errors from few getters before Ctx got cancled/timedout

Reply:
Ok, let's return aggregated. I am still not sure if there is any value from returning it when op is cancelled, but I want to see this in practice more.

Reply:
append the ctx.Err 👍🏻

Reply:
I assume the last getErr will be the context one so it will be already appended.

Reply:
We aggregate to see what errors happen before timeout happen. Aggregating context error also helps to identify what getter was in use when error happen. It will help us debug errors in cascade

Reply:
Also we collect error span in case context Error, which is also nice to have.